feat(nuxt): Improve Nuxt server-side setup docs (#12102)
getsentry/sentry-docs: docs/platforms/javascript/guides/nuxt/install/cli-import.mdx

## Understanding the `--import` CLI Flag

The [`--import` CLI flag](https://nodejs.org/api/cli.html#--importmodule) in Node is the default way in ESM to preload a module before the application starts.

> Setting this flag is crucial for ensuring proper server-side initialization, as Sentry needs to be initialized before the rest of the application runs.

Author: True, I changed it a bit: "Setting this flag is crucial for ensuring proper server-side initialization, as Sentry needs to be initialized before Node.js resolves the entire import graph of your application. This is necessary for enabling proper instrumentation of ESM module"

> Setting this flag is crucial for ensuring proper server-side initialization, as Sentry needs to be initialized before the rest of the application runs.

Author: After Lizas review, I took a look at this again and I think this is true for the dynamic import (as it creates a new import graph). But for `--import` it's actually correct, that it loads before the application starts.
However, I will update it in the dynamic import page, as I wrote the same sentence there.

From the Node docs:
- The hooks can be registered before the application code is run by using the `--import`. [here](https://nodejs.org/api/module.html#enabling)
- Preload the specified module at startup. [here](https://nodejs.org/api/cli.html#--importmodule)

## Scenarios where `--import` does not work

> - You're not able to add Node CLI flags or environment variables scoped to the function runtime, meaning these variables are not applied in other scopes, such as build time.

Reviewer: ```suggestion
- You're not able to add Node CLI flags or environment variables scoped to the function runtime, meaning these variables are not applied in other scopes, such as at build time.
```

Author: I think we shouldn't use abbreviations like e.g. but let's see what Liza has to say :)

Reviewer 2: @s1gr1d is right :) we avoid latin abbreviations

- You don't know the path to the Nuxt server folder in the build output

> - **Netlify** only allows scoped environment variables on its paid plans at this time
> - **Vercel** doesn't currently provide an option for scoping environment variables

Reviewer: ```suggestion
- On **Netlify**, it is as of now not possible to configure `--import` properly. 
- On **Vercel**, it is as of now not possible to configure `--import` properly.
```

should we just write it like this? 🤔  As the unknown path is also an issue there, in addition to the scopes variables stuff?

If any of those points apply to you, you cannot use the `--import` flag to initialize Sentry on the server-side.

> Check out the guide for using <PlatformLink to="/install/top-level-import/">a top-level import</PlatformLink> instead.

Reviewer: ```suggestion
Check out the guide for using <PlatformLink to="/install/limited-server-tracing/">limited server tracing</PlatformLink> instead.
```

## Understanding the `--import` CLI Flag

The [`--import` CLI flag](https://nodejs.org/api/cli.html#--importmodule) in Node is the default way in ESM to preload a module before the application starts.
Setting this flag is crucial for ensuring proper server-side initialization, as Sentry needs to be initialized before
Node.js resolves the entire import graph of your application. This is necessary for enabling proper instrumentation of ESM module

## Scenarios where `--import` does not work

> - You're not able to add Node CLI flags or environment variables scoped to the function runtime, meaning these variables are not applied in other scopes, such as build time.

Reviewer 2: ```suggestion
- You're unable to add Node CLI flags or environment variables scoped to the function runtime, meaning these variables aren't applied in other scopes, such as build time.
```

> - You don't know the path to the Nuxt server folder in the build output
> - On **Netlify**, it is as of now not possible to configure `--import` properly.
> - On **Vercel**, it is as of now not possible to configure `--import` properly.

Reviewer 2: ```suggestion
- You don't know the path to the Nuxt server folder in the build output
- At this time, it isn't possible to properly configure `--import` in **Netlify**.
- At this time, it isn't possible to properly configure `--import` in **Vercel**.
```
